src/wallet/wallet.cpp

```diff
@@ -1597,6 +1599,7 @@ void CWalletTx::GetAmounts(std::list<COutputEntry>& listReceived,
         nFee = nDebit - nValueOut;
     }
 
+    LOCK(pwallet->cs_wallet);
```

Comment:
In commit "wallet: IsMine overloads require cs_wallet lock" (d8441f30ff57e4ae98cff6694c995eaffc19c51a)

Maybe with this expanded lock scope. EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) could be added to IsChange(), and that method wouldn't need to lock cs_wallet recursively. Not sure how much extra work that is, however, so maybe it is a suggestion for a followup

Reply:
Will look into it.

Reply:
Pushed b8405b833ad28351c80fb10f6f896f974013fd9e which touches `IsChange` and `GetChange` only.